pkg/image/registry/imagestreamimage/rest.go

```diff
@@ -34,29 +33,12 @@ func (r *REST) New() runtime.Object {
 	return &api.ImageStreamImage{}
 }
 
-// ParseNameAndID splits a string into its name component and ID component, and returns an error
```

Comment:
By moving this you've changed the messages in an inconsistent way.  This method was explicitly used to give an error that made sense to an API user - if you change the code that uses it they'll lose that error.  A user accessing the api is expected to know about `ImageStreamImage`, a user calling other code is not.  So the messages in the image/api/helper need to be generic, the messages here need to be specific.  It would be better if you had this method (as private) wrapping and calling imageapi.Parse... but returned the existing message here on error.